### PR TITLE
Add missing Index API Spec Tests, Part 1

### PR DESCRIPTION
**Pull request by Tokesh**
### Description
Adding tests for this API list:
  POST /{index}/_mapping
  GET /{index}/_mapping/field/{fields}
  POST /{index}/_mget
  GET /{index}/_msearch
  POST /{index}/_msearch
  GET /{index}/_validate/query
  POST /{index}/_validate/query
### Issues Resolved
[#663 ]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Tokesh** [comment]
As soon as my PR with replication is merged, I will rebase it so that it doesn't interfere with checking
In the meantime, I made it a draft because I am still in the process of adding tests. I will try to do them as quickly as possible)

File: tests/default/indices/mapping.yaml [hunk 1]
```diff
@@ -146,3 +146,40 @@ chapters:
...
+        properties:
+          genre:
+            type: text
+          rating:
+            type: float
+    response:
+      status: 200
+      payload:
+        acknowledged: true
+  - synopsis: Get specific field mappings for an index.
```
Comment (dblock) [inline tests/default/indices/mapping.yaml:165]: Consider splitting tests for different paths into their own scenarios if there's no duplication, e.g. this one would be `tests/default/indices/mapping/field.yaml`.

File: tests/default/indices/msearch.yaml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    method: POST
+    parameters:
+      index: books
+      id: '3'
+    request:
+      payload:
+        name: Book 3
+        age: 30
+    response:
+      status: 201
```
Comment (dblock) [inline tests/default/indices/msearch.yaml:74]: These should be in a prologue. You can also use `_bulk` from other examples which makes it much shorter.

File: tests/default/indices/msearch.yaml [hunk 3]
```diff
@@ -0,0 +1,124 @@
+$schema: ../../../json_schemas/test_story.schema.yaml
+
+description: Test endpoints relevant to the lifecycle of an index, including multi-get and multi-search operations.
+epilogues:
+  - path: /books
+    method: DELETE
+    status: [200, 404]
+  - path: /games
```
Comment (dblock) [inline tests/default/indices/msearch.yaml:8]: No games index in this test, remove.

File: tests/default/indices/msearch.yaml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      payload:
+        mappings:
+          properties:
+            name:
+              type: keyword
+            age:
+              type: integer
+        settings:
+          number_of_shards: 5
+          number_of_replicas: 2
```
Comment (dblock) [inline tests/default/indices/msearch.yaml:27]: I don't think number of shards and replicas is material for this test, remove. If msearch works the same without a mapping, as well.

File: tests/default/indices/msearch.yaml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      index: books
+      id: '3'
+    request:
+      payload:
+        name: Book 3
+        age: 30
+    response:
+      status: 201
+
+  - synopsis: Perform a `_mget` request to retrieve documents by IDs.
```
Comment (dblock) [inline tests/default/indices/msearch.yaml:76]: This belongs in mget.yaml.

File: tests/default/indices/refresh.yaml [hunk 6]
```diff
@@ -37,3 +37,27 @@ chapters:
       payload:
         _shards:
           failed: 0
+  - synopsis: Adding one more document.
```
Comment (dblock) [inline tests/default/indices/refresh.yaml:40]: We're not trying to test the server, but the API schema. I don't think this adds anything as we already call `_refresh` above with the same results.

**Review by dblock — CHANGES_REQUESTED** [review]
Thanks, some asks below.

**Comment by Tokesh** [comment]
> Thanks, some asks below.

Thank you for review, I have resolved the conversations and added ```/{index}/_validate/query``` API specs

File: tests/default/indices/validate_query.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
+$schema: ../../../json_schemas/test_story.schema.yaml
```
Comment (dblock) [inline tests/default/indices/validate_query.yaml:1]: This file should be `validate/query.yaml`.

**Review by dblock — CHANGES_REQUESTED** [review]
One file name change, sorry to be annoying.

File: tests/default/indices/validate_query.yaml (hunk 7 above)
Comment (Tokesh) [reply tests/default/indices/validate_query.yaml]: no worries! i'm really interested in working with OpenSearch in the form of tests or writing specs to understand it more deeply.

**Review by dblock — APPROVED** [review]

**Pull request merged**